### PR TITLE
openpam/20190224: Added openpam recipe

### PR DESCRIPTION
**Pull request by klausholstjacobsen**
openpam/20190224

Added recipe

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/openpam/all/test_package/test_package.c [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+#include <security/openpam.h>
+#include <security/pam_constants.h>
+
+
+int main(void) {
+
+    int x = PAM_SUCCESS;
+
+    return EXIT_SUCCESS;
+}
```
Comment (uilianries) [inline recipes/openpam/all/test_package/test_package.c:13]:
```suggestion
#include <stdio.h>
#include <stdlib.h>
#include <security/pam_types.h>
#include <security/openpam.h>
#include <security/pam_constants.h>
#include <security/pam_appl.h>


int main(void) {
    pam_handle_t* pamh;
    struct pam_conv pamc;
    const char *user;

    pam_start("yes", user, &pamc, &pamh);
    pam_end(pamh, PAM_SUCCESS);

    return EXIT_SUCCESS;
}

```
Need to consume some function from the library to check runtime linkage.

File: recipes/openpam/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    def package(self):
+        autotools = Autotools(self)
+        autotools.install()
+        copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"), ignore_case=True)
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rm(self, "*.la", os.path.join(self.package_folder, "lib"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["pam"]
```
Comment (uilianries) [inline recipes/openpam/all/conanfile.py:72]:
```suggestion
        self.cpp_info.libs = ["pam"]
        self.cpp_info.system_libs = ["dl"]
```

**Review by uilianries — COMMENTED** [review]
Looking good, only small

**Review by RubenRBS — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

File: recipes/openpam/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+from conan.tools.gnu import Autotools, AutotoolsToolchain
+from conan.tools.layout import basic_layout
+import os
+
+
+required_conan_version = ">=1.54.0"
+
+class PackageConan(ConanFile):
+    name = "openpam"
+    description = "OpenPAM library"
```
Comment (jwillikers) [inline recipes/openpam/all/conanfile.py:14]: This description isn't the most helpful. Can you elaborate on what the library is / does. Usually the first sentence from the project's website will suffice.

File: recipes/openpam/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    license = "BSD-3-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://openpam.org/"
+    topics = ("pam")
```
Comment (jwillikers) [inline recipes/openpam/all/conanfile.py:18]:
```suggestion
    topics = ("authentication", "pam")
```

**Review by SSE4 — APPROVED** [review]

**Pull request merged**